km62/Yevlentiev_Mykhailo/laboratory1.sql

```diff
@@ -1,2 +1,134 @@
 -- LABORATORY WORK 1
 -- BY Yevlentiev_Mykhailo
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці та видаляти з них дані
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER evlentiev IDENTIFIED BY love_sql
+DEFAULT TABLESPACE "USERS" TO evlentiev
+TEMPERARY TABLESPACE "TEMP" TO evlentiev
+
+ALTER "CONNECT" TO evlentiev
+ALTER CREATE ANY TABLE TO evlentiev
+ALTER DROP ANY TABLE TO evlentiev
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент здає роботу викладачу.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE TABLE STUDENT(
+  sudent_name VARCHAR2(30)
+);
+ALTER TABLE STUDENT
+  ADD CONSTRAIN student_pk PRIMERY KEY (student_name)
+
+
+CREATE TABLE TASK(
+  task_name VARCHAR2(30)
+);
+
+ALTER TABLE STUDENT
+  ADD CONSTRAIN task_pk PRIMERY KEY (task_name)
+
+CREATE TABLE STUDENT_TASK(
+  task_name_fk VARCHAR2(30),
+  sudent_name_fk VARCHAR2(30),
+  marks NUMBER(100)
+);
+
+ALTER TABLE STUDENT_TASK
+  ADD CONSTRAIN student_task_pk PRIMERY KEY (task_name_fk,student_name_fk)
+  
+ALTER TABLE STUDENT_TASK
+  ADD CONSTRAIN student_fk FOREIN KEY (student_name) REFERENCES STUDENT  
+  
+ALTER TABLE STUDENT_TASK
+  ADD CONSTRAIN task_fk FOREIN KEY (task_name) REFERENCES TASK 
+  
```

Comment:
Task is done incorrectly. Another error when creating the internal key.
created an external key in the student_fk and task_fk inventory table, which refers to a table, such as STUDENT, based on columns that are not specified and must be specified. The same error with TASK.